### PR TITLE
Updated to latest Gfx

### PR DESCRIPTION
**Pull request by bvssvni**
- Removed derived traits on `Texture`
- Bumped to 0.8.0

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -26,19 +28,16 @@ pub enum Flip {
 }
 
 /// Represents a texture.
-#[derive(Clone, Debug, PartialEq)]
 pub struct Texture<R> where R: gfx::Resources {
-    handle: gfx::handle::Texture<R>
+    /// Pixel storage for texture.
+    pub surface: gfx::handle::Texture<R, R8_G8_B8_A8>,
+    /// View used by shader.
+    pub view: gfx::handle::ShaderResourceView<R, [f32; 4]>
```
Comment (kvark) [inline src/lib.rs]: Now this texture starts to make some sense :)

File: src/lib.rs [hunk 2]
```diff
@@ -126,24 +125,13 @@ impl<F, R> Rgba8Texture<F> for Texture<R>
...
-        };
-        let tex_handle = match factory.create_texture_static(tex_info, &memory) {
-            Ok(x) => x,
-            Err(err) => { return Err(err); }
-        };
-        if settings.get_generate_mipmap() {
-            factory.generate_mipmap(&tex_handle);
-        }
-        Ok(Texture { handle: tex_handle })
+        let tex_info = gfx::tex::Kind::D2(width, height,
```
Comment (kvark) [inline src/lib.rs]: strictly speaking, the `tex_info` variable name is not correct

File: src/lib.rs [hunk 3]
```diff
@@ -152,18 +140,20 @@ impl<F, R> Rgba8Texture<F> for Texture<R>
...
+        ).map_err(|err| err.into())
     }
 }
 
 impl<R> ImageSize for Texture<R> where R: gfx::Resources {
     #[inline(always)]
     fn get_size(&self) -> (u32, u32) {
-        let info = self.handle.get_info();
-        (info.width as u32, info.height as u32)
+        match self.surface.get_info().kind {
```
Comment (kvark) [inline src/lib.rs]: Alternatively, you could use `get_dimensions` method for convenience

**Comment by bvssvni** [comment]
Perhaps a good idea to wait merging until https://github.com/gfx-rs/gfx/issues/857 is fixed.

**Comment by bvssvni** [comment]
Opened https://github.com/PistonDevelopers/gfx_texture/issues/76 to not forget it, but can wait for the next Gfx version.

Time to move on. Merging.

**Pull request merged**